ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -696,11 +682,15 @@ void open(CompactionInfo ci) throws TException {
             + "}, status {" + res.getState() + "}, reason {" + res.getErrorMessage() + "}");
       }
       lockId = res.getLockid();
-
-      heartbeatExecutor = Executors.newSingleThreadScheduledExecutor();
+      heartbeatExecutor = Executors.newSingleThreadScheduledExecutor(
+              CompactorUtil.createThreadFactory("CompactionTxn Heartbeater %d"));
```

Comment:
what would be the id, would it be compactionTxn.getTxnId() ?

Reply:
No, that is a builtin feature, it will be a sequential number assigned by the thread factory